extensions/amp-base-carousel/1.0/scroller.js

```diff
@@ -231,24 +200,25 @@ function ScrollerWithRef(
     if (!container) {
       return;
     }
-    if (mixedLength) {
-      const acc = {index: 0, width: 0};
-      for (let i = 0; i < container.children.length; i++) {
-        const slide = container.children[i];
-        if (container./* OK */ scrollLeft >= acc.width) {
-          scrollOffset.current = container./* OK */ scrollLeft - acc.width;
-          acc.width += slide./* OK */ scrollWidth;
-          acc.index = loop ? restingIndex - pivotIndex + i : i;
-        }
-      }
-      currentIndex.current = mod(acc.index, children.length);
-    } else {
-      const slideWidth = container./* OK */ offsetWidth / visibleCount;
-      scrollOffset.current =
-        container./* OK */ scrollLeft - offsetRef.current * slideWidth;
-      const slideOffset = Math.round(scrollOffset.current / slideWidth);
-      currentIndex.current = mod(slideOffset, children.length);
+    const overlappingIndex = findOverlappingIndex(
```

Comment:
The `findOverlappingIndex` is somewhat heavy. But I think it's still true that in a non-mixedLength case, all the math is drastically simpler since we don't actually need to know dimensions of each slide?

Reply:
Agreed. It's not ideal, but it only measures (calls `overlaps`) once on new renders when scrolling forward, which occurs on debounced scroll. When you scroll backwards it typically enters the for loop once, because the starting index is one away from the new overlap. In this case I honestly prefer the readability and maintainability.

Reply:
How does this work with possible carousel resizes? E.g. if window gets bigger?

Reply:
I'm not sure what you mean. Each measure includes a fresh call to `getBoundingClientRect`, so this should be updated after resize?

Reply:
Got it. So this is transactional only, right?